### DISCUSSION TITLE
Installation: I do not find the Open3Eclient

### DISCUSSION THREAD

**Discussion by docolli** [open3e/open3e#118] [Q&A] [answered]
Installed open3e into a virtual environment into folder ~/open3e:

```sh
python3 -m venv ~/open3e
cd ~/open3e
source ~/open3e/bin/activate
pip install git+https://github.com/open3e/open3e.git
```

`Open3E_depictSystem` runs fine and also with the command `open3e` I can grab data from my heatpump Vitocal 250-A.
However, I try to install open3e as a service according to your wiki. There it is said, that I can use a file with args and the command `Open3Eclient.py`. However, I do not find the python script `Open3Eclient.py` or the example template `args` on my system.

```
ls -l bin/
insgesamt 64
-rw-r--r-- 1 olli olli 1994 27. Aug 07:32 activate
-rw-r--r-- 1 olli olli  920 27. Aug 07:32 activate.csh
-rw-r--r-- 1 olli olli 2200 27. Aug 07:32 activate.fish
-rw-r--r-- 1 olli olli 9033 27. Aug 07:32 Activate.ps1
-rwxr-xr-x 1 olli olli  228 27. Aug 07:33 can_logconvert
-rwxr-xr-x 1 olli olli  224 27. Aug 07:33 can_logger
-rwxr-xr-x 1 olli olli  224 27. Aug 07:33 can_player
-rwxr-xr-x 1 olli olli  224 27. Aug 07:33 can_viewer
-rwxr-xr-x 1 olli olli  233 27. Aug 07:33 open3e
-rwxr-xr-x 1 olli olli  240 27. Aug 07:33 open3e_depictSystem
-rwxr-xr-x 1 olli olli  237 27. Aug 07:33 open3e_dids2json
-rwxr-xr-x 1 olli olli  236 27. Aug 07:32 pip
-rwxr-xr-x 1 olli olli  236 27. Aug 07:32 pip3
-rwxr-xr-x 1 olli olli  236 27. Aug 07:32 pip3.11
lrwxrwxrwx 1 olli olli    7 27. Aug 07:31 python -> python3
lrwxrwxrwx 1 olli olli   16 27. Aug 07:31 python3 -> /usr/bin/python3
lrwxrwxrwx 1 olli olli    7 27. Aug 07:31 python3.11 -> python3
```

Did I miss something for installation, or do I have to copy these files manually from the repository?

**Comment by philippoo66** [comment]
hi Olli @docolli  !

as far as I remember, using pip install the 'sources' get installed in /usr/lib/open3e/ or /usr/bin/open3e. If you wouldn't find it there just search the file system for open3e. must be somewhere ;-)

sudo find / -name "\*open3e\*.*"

do *not* copy the files manually somewehre - will cause 'chaos' ;-)

**Reply by MyHomeMyData** [reply]
@docolli You used venv installation, so open3e files should be located at `~/open3e/lib/python3.12/site-packages/open3e`

@SanMiggel Maybe you could give better advice? Do we have to adapt readme?

**Reply by docolli** [reply]
Thank you! Yes indeed, I've found it under the path you told me!

@SanMiggel The readme says that the command `open3e` can be used with `@args`, that is clear. However, I was following the wiki and there only `Open3Eclient.py` is mentioned. Also in the section how to run it as a service. Maybe this wiki section has to be updated.

**Comment by SanMiggel** [comment, marked as answer]
The best way would be to activate the virtual env in the service script. Then all the paths will be set accordingly.

This you can simply use
> open3e ...

I think this should be clear from the readme.



-------- Original message --------
From: MyHomeMyData ***@***.***>
Date: 27/08/2024 09:57 (GMT+01:00)
To: open3e/open3e ***@***.***>
Cc: SanMiggel ***@***.***>, Mention ***@***.***>
Subject: Re: [open3e/open3e] Installation: I do not find the Open3Eclient (Discussion #118)


@docolli<https://github.com/docolli> You used venv installation, so open3e files should be located at ~/open3e/lib/python3.12/site-packages/open3e

@SanMiggel<https://github.com/SanMiggel> Maybe you could give better advice? Do we have to adapt readme?

**Comment by SanMiggel** [comment]
Ah right. We should update the wiki. Will try do do it soon.

-------- Original message --------
From: docolli ***@***.***>
Date: 27/08/2024 18:50 (GMT+01:00)
To: open3e/open3e ***@***.***>
Cc: SanMiggel ***@***.***>, Mention ***@***.***>
Subject: Re: [open3e/open3e] Installation: I do not find the Open3Eclient (Discussion #118)


Thank you! Yes indeed, I've found it under the path you told me!

@SanMiggel<https://github.com/SanMiggel> The readme says that the command open3e can be used with @args, that is clear. However, I was following the wiki and there only Open3Eclient.py is mentioned. Also in the section how to run it as a service. Maybe this wiki section has to be updated.
